main.go

```diff
@@ -115,16 +123,49 @@ func main() {
 		os.Exit(1)
 	}
 
+	factory := conditions.InClusterFactory{Client: mgr.GetClient()}
```

Comment:
Pls create the func for this block of code.

Reply:
no qualms, will do.